examples/nlp/text_classification_with_switch_transformer.py

```diff
@@ -0,0 +1,324 @@
+        # expert_outputs_combined shape: [tokens_per_batch, embed_dim]
+        expert_outputs_combined = tf.einsum(
+            "abc,xba->xc", expert_outputs, combine_tensor
+        )
+        # output shape: [batch_size, num_tokens_per_example, embed_dim]
+        outputs = tf.reshape(
+            expert_outputs_combined,
+            [batch_size, num_tokens_per_example, self.embed_dim],
+        )
+        return outputs
+
+
+"""
+## Implement a Transformer block as a layer
+"""
+
+
+class TransformerBlock(layers.Layer):
+    def __init__(self, embed_dim, num_heads, ffn, dropout_rate=0.1):
+        super(TransformerBlock, self).__init__()
+        self.att = layers.MultiHeadAttention(num_heads=num_heads, key_dim=embed_dim)
+        # The ffn can be either a standard feedforward network or a switch
+        # layer with a Mixture of Experts.
+        self.ffn = ffn
+        self.layernorm1 = layers.LayerNormalization(epsilon=1e-6)
+        self.layernorm2 = layers.LayerNormalization(epsilon=1e-6)
+        self.dropout1 = layers.Dropout(dropout_rate)
+        self.dropout2 = layers.Dropout(dropout_rate)
+
+    def call(self, inputs, training):
+        attn_output = self.att(inputs, inputs)
+        attn_output = self.dropout1(attn_output, training=training)
+        out1 = self.layernorm1(inputs + attn_output)
+        ffn_output = self.ffn(out1)
+        ffn_output = self.dropout2(ffn_output, training=training)
+        return self.layernorm2(out1 + ffn_output)
+
+
+"""
+## Implement the classifier
+
+Transformer layer outputs one vector for each time step of our input sequence.
+Here, we take the mean across all time steps and use a feedforward network on top
+of it to classify text.
+"""
+
+
+def create_classifier():
+
+    switch = Switch(num_experts, embed_dim, num_tokens_per_batch)
+    transformer_block = TransformerBlock(ff_dim, num_heads, switch)
+
+    inputs = layers.Input(shape=(num_tokens_per_example,))
+    embedding_layer = TokenAndPositionEmbedding(
+        num_tokens_per_example, vocab_size, embed_dim
+    )
+    x = embedding_layer(inputs)
+    x = transformer_block(x)
+    x = layers.GlobalAveragePooling1D()(x)
+    x = layers.Dropout(dropout_rate)(x)
+    x = layers.Dense(ff_dim, activation="relu")(x)
+    x = layers.Dropout(dropout_rate)(x)
+    outputs = layers.Dense(2, activation="softmax")(x)
+
+    classifier = keras.Model(inputs=inputs, outputs=outputs)
+    return classifier
+
+
+"""
+## Train and evaluate the model
+"""
+
+
+def run_experiment(classifier):
+    classifier.compile(
+        optimizer=keras.optimizers.Adam(learning_rate),
+        loss="sparse_categorical_crossentropy",
+        metrics=["accuracy"],
+    )
+
+    history = classifier.fit(
+        x_train,
+        y_train,
+        batch_size=batch_size,
+        epochs=num_epochs,
+        validation_data=(x_val, y_val),
+    )
+    return history
+
+
+classifier = create_classifier()
+run_experiment(classifier)
```

Comment:
We'd need some final display of test results and comment about these results

Reply:
- We provide a validation data in the `fit` method , for which accuracy is displayed during training. I didn't add the final display for this split as you suggested in the other example.
- I added a "general" conclusion text, but not commenting on the produced results as we are not comparing it with the standard Transformer. 

What do you think?